config/routes.rb

```diff
@@ -14,7 +14,7 @@ def show_step(name)
 end
 
 Rails.application.routes.draw do
-  devise_for :users
+  devise_for :users, only: :passwords
```

Comment:
I've realised in theory it should be possible to specify our custom controllers and to not have to override the authenticate_user! method or any other:

https://github.com/plataformatec/devise/blob/master/lib/devise/rails/routes.rb#L131

Although not sure that would be enough.